### ISSUE TITLE
Unexpected end of file when returning specs.4.8.gz

### ISSUE THREAD

**Issue by joshleblanc** [geminabox/geminabox#354]
This is a new problem that started recently. Previously geminabox worked without a hitch. Basically when trying to fetch the spec files, geminabox throws an error and returns 500.

I've tried to delete the specs files and re-generate them, however the problem persists. The problem exists for `latest_specs.4.8.gz` and `specs.4.8.gz`, however `prerelease_specs4.8.gz` downloads fine.

I can manually decompress both gzip files with 7zip without issue.

Is there anything else I can try to resolve this issue?

```
2021-06-08 09:03:03 - Zlib::GzipFile::Error - unexpected end of file:
        C:/Ruby27-x64/lib/ruby/2.7.0/rubygems/util.rb:23:in `read'
        C:/Ruby27-x64/lib/ruby/2.7.0/rubygems/util.rb:23:in `gunzip'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/splicer.rb:69:in `unpackage'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/splicer.rb:62:in `merge_gziped_content'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/splicer.rb:49:in `merge_content'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/splicer.rb:29:in `new_content'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/splicer.rb:16:in `create'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/splicer.rb:11:in `make'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/hostess.rb:70:in `splice_file'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/geminabox-1.4.1/lib/geminabox/proxy/hostess.rb:21:in `block (2 levels) in<class:Hostess>'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1675:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1675:in `block in compile!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1013:in `block (3 levels) in route!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1032:in `route_eval'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1013:in `block (2 levels) in route!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1061:in `block in process_route'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1059:in `catch'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1059:in `process_route'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1011:in `block in route!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1008:in `each'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1008:in `route!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1129:in `block in dispatch!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `block in invoke'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `catch'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `invoke'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1124:in `dispatch!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:939:in `block in call!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `block in invoke'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `catch'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `invoke'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:939:in `call!'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:929:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/xss_header.rb:18:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/path_traversal.rb:16:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/json_csrf.rb:26:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/frame_options.rb:31:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/null_logger.rb:11:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:216:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1991:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/xss_header.rb:18:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/path_traversal.rb:16:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/json_csrf.rb:26:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/frame_options.rb:31:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/null_logger.rb:11:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/method_override.rb:24:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:216:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1991:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1542:in `block in call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1769:in `synchronize'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1542:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/puma-5.3.2/lib/puma/configuration.rb:249:in `call'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/puma-5.3.2/lib/puma/request.rb:77:in `block in handle_request'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/puma-5.3.2/lib/puma/thread_pool.rb:338:in `with_force_shutdown'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/puma-5.3.2/lib/puma/request.rb:76:in `handle_request'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/puma-5.3.2/lib/puma/server.rb:438:in `process_client'
        C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/puma-5.3.2/lib/puma/thread_pool.rb:145:in `block in spawn_thread'
```

**Comment by ahorek** [comment]
I'm not 100% sure, but I think that https://github.com/geminabox/geminabox/pull/357 should fix this problem as well.

**Comment by joshleblanc** [comment]
That definitely looks like it would potentially be the cause

**Comment by tnir** [comment]
@joshleblanc As I just released, would you mind trying out geminabox 2.0.0 which includes @ahorek's fix #357? cc @ahorek.
